src/main/java/spring/rest/task/repository/EmailRepository.java

```diff
@@ -0,0 +1,59 @@
+package spring.rest.task.repository;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.stereotype.Repository;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class EmailRepository {
+    @Value("${email.filename}")
+    private String filename;
+    private List<SimpleMailMessage> emailList;
+
+    public EmailRepository(){
+        this.emailList = new ArrayList<>();
+    }
+    public List<SimpleMailMessage> findAll(){
+        if(!Files.exists(Paths.get(String.valueOf(filename)))){
+            try {
+                Files.createFile(Paths.get(String.valueOf(filename)));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(String.valueOf(filename)))) {
+            emailList = (List<SimpleMailMessage>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return emailList;
+    }
+    public void deleteAll(){
+        emailList.clear();
+        save();
+    }
+    public void delete(SimpleMailMessage simpleMailMessage){
+        emailList.remove(simpleMailMessage);
+        save();
+    }
+    public void save(){
+        try (ObjectOutputStream ous = new ObjectOutputStream(new FileOutputStream(String.valueOf(filename)))) {
+            ous.writeObject(emailList);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    public void save(List<SimpleMailMessage> simpleMailMessages){
```

Comment:
You can postpone save to the moment when application stops. I/O operation is slower than in memory. Therefore while application is running you should work with in memory representation of your data - `emailList` collection you have